test/test_grid_file.py

```diff
@@ -675,6 +675,23 @@ def test_context_manager(self):
         with GridOut(self.db.fs, infile._id) as outfile:
             self.assertEqual(contents, outfile.read())
 
+    def test_exception_file_non_existence(self):
+        contents = b"Imagine this is some important data..."
+
+        try:
+            with GridIn(self.db.fs, filename="important") as infile:
+                infile.write(contents)
+                raise CorruptGridFile("Test exception")
```

Comment:
CorruptGridFile would never actually be raised by GridIn.write so it probably makes more sense to use pymongo.errors.ConnectionFailure or something.